src/Files.App/Actions/Show/TogglePreviewPaneAction.cs

```diff
@@ -0,0 +1,39 @@
+﻿using CommunityToolkit.Mvvm.ComponentModel;
+using Files.App.Commands;
+using Files.App.Extensions;
+using Files.App.ViewModels;
+using System.ComponentModel;
+using System.Threading.Tasks;
+using Windows.System;
+
+namespace Files.App.Actions
+{
+	internal class TogglePreviewPaneAction : ObservableObject, IToggleAction
+	{
+		private readonly PreviewPaneViewModel viewModel = App.PreviewPaneViewModel;
+
+		public string Label { get; } = "TogglePreviewPane".GetLocalizedResource();
+
+		public RichGlyph Glyph { get; } = new(opacityStyle: "ColorIconRightPane");
+		public HotKey HotKey { get; } = new(VirtualKey.P, VirtualKeyModifiers.Control);
+
+		public bool IsOn => viewModel.IsEnabled;
+
+		public TogglePreviewPaneAction()
+		{
+			viewModel.PropertyChanged += ViewModel_PropertyChanged;
+		}
+
+		public Task ExecuteAsync()
+		{
+			viewModel.IsEnabled = !IsOn;
```

Comment:
Shouldn't we check if ViewModel is not null ?

Reply:
This property is defined only once when launching the application. It cannot be null.